tests/rptest/tests/topic_delete_test.py

```diff
@@ -73,6 +73,94 @@ def topic_storage_purged():
             raise
 
 
+class TopicDeleteCloudStorageTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3,
+                        cleanup_policy=TopicSpec.CLEANUP_DELETE), )
+
+    def __init__(self, test_context):
+        self.si_settings = SISettings(log_segment_size=1024 * 1024)
+        super().__init__(test_context=test_context,
+                         num_brokers=3,
+                         si_settings=self.si_settings)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+
+    @cluster(num_nodes=3)
+    @parametrize(disable_delete=False)
+    @parametrize(disable_delete=True)
+    def topic_delete_cloud_storage_test(self, disable_delete):
+        # Set retention to 5MB
+        self.kafka_tools.alter_topic_config(
+            self.topic, {'retention.bytes': 5 * 1024 * 1024})
```

Comment:
`SISettings` enables remote write at the cluster level by default. To adjust the local retention you have to set `redpanda.remote.local.target.bytes`.